### DISCUSSION TITLE
Help running tunarr

### DISCUSSION THREAD

**Discussion by jasongdove** [chrisbenincasa/tunarr#256] [General]
I'm trying to run tunarr, but unfortunately not having any luck.

I see the readme says 

>  ## Development
> 
> [pnpm](https://pnpm.io) and [turbo](https://turbo.build/) are used for package management and development. More details on the development environment, setup, and contributing to come!
> 
> ### Start dev servers
> 
> Run from the root of the project:
> 
> ```
> pnpm turbo dev
> ```

When I do `pnpm install && pnpm turbo dev` that seems to start up the server/api, but I can't figure out how to access the UI.

I also tried Docker using `docker build -t tunarr -f build.Dockerfile .`, but it looks like those have fallen out of date

> [prod-deps 1/1] RUN --mount=type=cache,id=pnpm,target=/pnpm/store pnpm install --prod --frozen-lockfile:
> 0.578 Corepack is about to download https://registry.npmjs.org/pnpm/-/pnpm-8.15.4.tgz.
> 1.191 Scope: all 5 workspace projects
> 1.265  ENOENT  ENOENT: no such file or directory, open '/tunarr/patches/ts-essentials@9.4.1.patch'
> 1.265
> 1.265 pnpm: ENOENT: no such file or directory, open '/tunarr/patches/ts-essentials@9.4.1.patch'

When I add `COPY patches ./patches` to get past that, it then fails with

> 8.353 . prepare$ husky
> 8.362 . prepare: sh: husky: not found
> 8.364  ELIFECYCLE  Command failed.

Admittedly, this is not a stack that I use, so any direction you could provide would be helpful.

**Comment by chrisbenincasa** [comment]
Hi! Sorry that you're having difficulty. How have you tried accessing the web app after running `pnpm turbo dev`? When started that way, the server should be accessible on `localhost:8000` and the webdev server on `localhost:5173/web`. The logs displaying the latter may have gotten obliterated by turbo on startup...

As for the docker images -- yes it looks like I missed updating those when I had to add a custom patch for ts-essentials when updating to Typescript 5.4.3. I'm going to fix up the Dockerfile now. I'm also working on a github action to periodically release a 'latest' image of Tunarr hourly. Will keep you updated!

**Comment by jasongdove** [comment]
Thanks, port 5173 works - yeah, I didn't see that in the output anywhere, just 8000.

**Comment by chrisbenincasa** [comment]
Yeah, my guess is that there's a console.clear somewhere in the library code and that blows it away. I can also update the README to mention the default ports / paths for the dev servers.

**Comment by oneingan** [comment]
I will share with you soon(TM) a PR to develop, build and deploy using Nix.

**Comment by chrisbenincasa** [comment]
Wow awesome! Thanks!
